src/convertToObject.js

```diff
@@ -3,16 +3,33 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles (see an example in
+ * [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
   // write your code here
+  return sourceString
+    .split(';')
+    .map(property => {
+      return property
+        .split(':')
+        .map(element => element.trim())
+        .filter(element => element !== '');
+    })
+    .filter(empty => empty.length)
+    .sort(([a], [b]) => a.localeCompare(b))
+    .reduce((styles, currentProperty) => {
```

Comment:
Try [array destructuring](https://www.freecodecamp.org/news/array-destructuring-in-es6-30e398f21d10/) here, you will avoid using mysterious values like property[0]